pkg/controllers/localvolumeset/localvolumeset_controller.go

```diff
@@ -127,17 +137,38 @@ func (r *LocalVolumeSetReconciler) syncStorageClass(ctx context.Context, lvs *lo
 			Labels: map[string]string{
 				common.OwnerNameLabel:      lvs.GetName(),
 				common.OwnerNamespaceLabel: lvs.GetNamespace(),
+				common.OwnerKindLabel:      localv1alpha1.LocalVolumeSetKind,
 			},
 		},
 		Provisioner:       "kubernetes.io/no-provisioner",
 		ReclaimPolicy:     &deleteReclaimPolicy,
 		VolumeBindingMode: &firstConsumerBinding,
 	}
+	currentStorageClass := &storagev1.StorageClass{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Name: lvs.Spec.StorageClassName}, currentStorageClass); err != nil {
+		if kerrors.IsNotFound(err) {
+			err := r.Client.Create(ctx, storageClass)
+			if err != nil && !kerrors.IsAlreadyExists(err) {
+				return err
+			}
```

Comment:
I think you should return nil here if the storageclass was created successfully.
In that case, `currentStorageClass` could be empty and there is no need to call `r.Client.Update()` if it was just created.
Really any case where `r.Client.Get()` returns an error could leave `currentStorageClass` empty and may even tigger a panic on line 159 when checking the labels, so we should return before then.

Reply:
Good catch, thanks!